core/src/main/java/org/sonarsource/sonarlint/core/container/storage/SonarQubeActiveRulesProvider.java

```diff
@@ -21,57 +21,57 @@
 
 import java.util.ArrayList;
 import java.util.Collection;
+import java.util.Collections;
 import java.util.List;
 import java.util.Map;
+import java.util.function.Function;
 import java.util.stream.Collectors;
 import org.picocontainer.injectors.ProviderAdapter;
 import org.sonar.api.batch.rule.ActiveRules;
 import org.sonar.api.resources.Languages;
 import org.sonarsource.sonarlint.core.client.api.common.Language;
 import org.sonarsource.sonarlint.core.client.api.connected.ConnectedAnalysisConfiguration;
 import org.sonarsource.sonarlint.core.client.api.connected.ConnectedGlobalConfiguration;
-import org.sonarsource.sonarlint.core.client.api.exceptions.StorageException;
 import org.sonarsource.sonarlint.core.commons.log.SonarLintLogger;
 import org.sonarsource.sonarlint.core.container.analysis.SonarLintRules;
 import org.sonarsource.sonarlint.core.container.global.DefaultActiveRules;
 import org.sonarsource.sonarlint.core.container.standalone.rule.StandaloneActiveRuleAdapter;
 import org.sonarsource.sonarlint.core.container.standalone.rule.StandaloneRule;
-import org.sonarsource.sonarlint.core.serverapi.qualityprofile.QualityProfile;
 import org.sonarsource.sonarlint.core.serverapi.rules.ServerRules;
+import org.sonarsource.sonarlint.core.storage.ProjectStorage;
+import org.sonarsource.sonarlint.core.storage.RuleSet;
 
 public class SonarQubeActiveRulesProvider extends ProviderAdapter {
 
   private static final SonarLintLogger LOG = SonarLintLogger.get();
 
   private ActiveRules activeRules;
 
-  public ActiveRules provide(RulesStore rulesStore, List<QualityProfile> qProfiles, StorageReader storageReader, SonarLintRules rules,
-    ConnectedAnalysisConfiguration analysisConfiguration, Languages languages, ConnectedGlobalConfiguration globalConfiguration, ActiveRulesStore activeRulesStore) {
-    if (activeRules == null) {
-
-      Map<String, String> qProfilesByLanguage = loadQualityProfilesFromStorage(qProfiles, storageReader, analysisConfiguration);
+  public ActiveRules provide(SonarLintRules rules, RulesStore rulesStore, ConnectedAnalysisConfiguration analysisConfiguration,
+    Languages languages, ConnectedGlobalConfiguration globalConfiguration, ProjectStorage projectStorage) {
 
+    if (activeRules == null) {
       Collection<org.sonar.api.batch.rule.ActiveRule> activeRulesList = new ArrayList<>();
-      for (Map.Entry<String, String> entry : qProfilesByLanguage.entrySet()) {
-        String language = entry.getKey();
-        if (languages.get(language) == null) {
-          continue;
-        }
-
-        String qProfileKey = entry.getValue();
-        QualityProfile qProfile = qProfiles.stream().filter(qp -> qp.getKey().equals(qProfileKey)).findFirst().orElseThrow();
+      // could be empty before the first sync
+      var projectKey = analysisConfiguration.projectKey();
+      if (projectKey == null) {
+        // XXX should not happen, how to handle this case ?
```

Comment:
I think we can stop supporting analysis with default quality profiles (= no project key)